packages/account/src/Sections/Profile/PhoneVerification/confirm-phone-number.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState, useEffect, ChangeEvent } from 'react';
+import PhoneVerificationCard from './phone-verification-card';
+import { Button, Snackbar, Text, TextFieldAddon } from '@deriv-com/quill-ui';
+import { Localize, localize } from '@deriv/translations';
+import { observer, useStore } from '@deriv/stores';
+import { usePhoneNumberVerificationSetTimer, useRequestPhoneNumberOTP, useSettings } from '@deriv/hooks';
+import { VERIFICATION_SERVICES } from '@deriv/shared';
+import { validatePhoneNumber } from './validation';
+
+type TConfirmPhoneNumber = {
+    setOtpVerification: (value: { show_otp_verification: boolean; phone_verification_type: string }) => void;
+};
+
+const ConfirmPhoneNumber = observer(({ setOtpVerification }: TConfirmPhoneNumber) => {
+    const [phone_number, setPhoneNumber] = useState('');
+    const [phone_verification_type, setPhoneVerificationType] = useState('');
+    const [is_button_loading, setIsButtonLoading] = useState(false);
+    const {
+        requestOnSMS,
+        requestOnWhatsApp,
+        error_message,
+        setErrorMessage,
+        setUsersPhoneNumber,
+        is_email_verified,
+        email_otp_error,
+    } = useRequestPhoneNumberOTP();
+    const { data: account_settings, invalidate } = useSettings();
```

Comment:
is invalidate exported from the hook?

Reply:
yes @yauheni-deriv it's being exported from the hook

Reply:
I don't see it returning from the hook.
this is looking weird, we have separate hook to invalidate query. what is the use for returning it from custom hook and call invalidate with manually passed key?

Reply:
I've designed the useSetting hooks based on our existing useSetting from api/hooks. i think this invalidate will help to rerender the current useSettings hook and retrigger data from useSettings. If we use it separately, i think data will not be trigger from useSettings 🤔